rtos/TARGET_CORTEX/mbed_rtx_handlers.c

```diff
@@ -107,29 +107,41 @@ static const char *error_msg(int32_t status)
     }
 }
 
+static void trap_rtx_error(int32_t rtx_status, mbed_error_status_t error_status)
+{
+    // Attempts to get the console for the first time while printing an error
+    // may well cause a mutex error; in general let RTX calls fail during
+    // an error condition.
+    if (mbed_get_error_in_progress()) {
+        return;
+    }
+    MBED_ERROR1(error_status, error_msg(rtx_status), rtx_status);
+}
+
 void EvrRtxKernelError(int32_t status)
```

Comment:
@kjbracey-arm - It appears we are going backwards on the error information we are able to collect in these scenarios - like the IDs of the objects. Can't we pass that into trap_rtx_error() to be called in MBED_ERROR1 instead of status and rtx_status which is sort of redundant info?

Reply:
Ah, my mistake. I was just copy-pasting from `EvrRtxKernelError`, and didn't spot the others were passing the object ID as `error_status` instead.